pilot/pkg/networking/plugin/authn/util.go

```diff
@@ -0,0 +1,29 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package authn
+
+import (
+	meshconfig "istio.io/api/mesh/v1alpha1"
+	"istio.io/istio/pilot/pkg/features"
+)
+
+func trustDomainsForValidation(meshConfig *meshconfig.MeshConfig) []string {
+	var trustDomains []string
+	tdAliases := meshConfig.TrustDomainAliases
```

Comment:
this doesn't really solve the issue I mentioned, can you do it like the following:
```go
func trustDomainsForValidation(meshConfig *meshconfig.MeshConfig) []string {
  if features.SkipValidateTrustDomain.Get() {
    return nil
  }
  return append([]string{meshConfig.TrustDomain}, meshConfig.TrustDomainAliases)
}
```

Reply:
Oh I misunderstood you, I'll push a fix for this